### PR TITLE
also optimize objects deletion on removing instance and cli

### PR DESCRIPTION
**Pull request by foxriver76**

### CODE REVIEW

File: packages/cli/lib/cli/cliObjects.js [hunk 1]
```diff
@@ -348,10 +348,19 @@ module.exports = class CLIObjects extends CLICommand {
         if (!ids || !ids.length) {
             return tools.maybeCallback(callback);
         } else {
+            let allEnums;
+
+            try {
+                // cache all enums, else it will be slow to delete many objects
+                allEnums = tools.getAllEnums(objects);
```
Comment (Apollon77) [inline packages/cli/lib/cli/cliObjects.js:355]:
```suggestion
                allEnums = await tools.getAllEnums(objects);
```

File: packages/cli/lib/cli/cliObjects.js [hunk 2]
```diff
@@ -348,10 +348,19 @@ module.exports = class CLIObjects extends CLICommand {
...
+            } catch (e) {
+                console.error(`Could not cache enums: ${e.message}`);
```
Comment (Apollon77) [inline packages/cli/lib/cli/cliObjects.js:357]:
```suggestion
                console.error(`Could not retrieve all enums: ${e.message}`);
```

File: packages/controller/lib/setup/setupInstall.js [hunk 3]
```diff
@@ -1247,6 +1247,17 @@ function Install(options) {
...
         }
 
+        let allEnums;
+
+        if (objIDs.length > 1) {
+            try {
+                // cache all enums, else it will be slow to delete many objects
+                allEnums = await tools.getAllEnums(objects);
+            } catch (e) {
+                console.error(`host.${hostname}: Could not cache enums: ${e.message}`);
```
Comment (Apollon77) [inline packages/controller/lib/setup/setupInstall.js:1257]:
```suggestion
                console.error(`host.${hostname}: Could not retrieve all enums: ${e.message}`);
```

**Review by Apollon77 — CHANGES_REQUESTED** [review]

File: packages/cli/lib/cli/cliObjects.js (hunk 1 above)
Comment (foxriver76) [reply packages/cli/lib/cli/cliObjects.js:355]: ooops ;-)

**Pull request merged**